### ISSUE TITLE
Running the script only after it is loaded

### ISSUE THREAD

**Issue by serg-pervushin** [wp-media/wp-rocket#3208]
Hello.

The delayed upload feature is great! But often the script starts running before it has loaded. For example, the script is enabled maps.js, and below is the executable inline js, which should run only after loading the main one maps.js. Adding inline js to the deferred upload field doesn't help, it still runs faster than the main file. It would be great to set the priority of files so that they are loaded one at a time, and not all at once (or maybe set some dependencies).

Sorry if something is unclear. :)

**Comment by Garnet-Fox** [comment]
This behavior is noticed not only in inline, but in any delayed scripts, they are loaded sequentially, and execution is faster for those who are smaller, i.e. full async. There is also a question about delayed scripts, when exactly is the launch load relatively of the defered scripts on time or strictly after?
Sorry for the bad english.

**Comment by jazir555** [comment]
> This behavior is noticed not only in inline, but in any delayed scripts, they are loaded sequentially, and execution is faster for those who are smaller, i.e. full async :(. There is also a question about delayed scripts, when exactly is the launch load relatively of the defered scripts on time or strictly after?
> Sorry for the bad english.

This may explain why sometimes files appear to load out of order. I have preload setup for fonts, and sometimes one of my js files is loaded before the font file(even though it's preloaded). This actually has a big performance impact for me when it occurs, but it seems to do so spontaneously

**Comment by MuhammedSwalihct** [comment]
I am facing this issue as well... When we put two scripts that depend on each other, some times it works without any issues. But sometimes, it breaks things with the error "example... function is not defined" because the other script hasn't been loaded yet. Does anyone have a fix for this?

**Comment by GeekPress** [comment]
With our new Delay JS option scheduled for 3.9, this problem will be automatically fixed. So I'm closing this issue.
